axum-core/src/extract/rejection.rs

```diff
@@ -28,12 +29,74 @@ impl fmt::Display for BodyAlreadyExtracted {
 
 impl std::error::Error for BodyAlreadyExtracted {}
 
-define_rejection! {
-    #[status = BAD_REQUEST]
-    #[body = "Failed to buffer the request body"]
-    /// Rejection type for extractors that buffer the request body. Used if the
-    /// request body cannot be buffered due to an error.
-    pub struct FailedToBufferBody(Error);
+/// Rejection type for extractors that buffer the request body. Used if the
+/// request body cannot be buffered due to an error.
+// TODO: in next major for axum-core make this a #[non_exhaustive] enum so we don't need the
+// additional indirection
```

Comment:
Couldn't we do this now? The only thing I know you can do with the current definition you couldn't with an enum is matching on it with a pattern of `FailedToBufferBody { .. }` and probably `FailedToBufferBody(..)`, which is a useless operation that might be disallowed entirely in the future to make changes from struct-with-only-private-fields to enum strictly okay.

Reply:
I think you're right 😊 I've changed it to an enum.